flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java

```diff
@@ -128,9 +128,37 @@ public class FlinkContainers implements BeforeAllCallback, AfterAllCallback {
     @Nullable private RestClusterClient<StandaloneClusterId> restClusterClient;
     private boolean isStarted = false;
 
+    // TODO: update Class JavaDocs when design is fixed
+    /** The FlinkContainers builder. */
+    public static final class Builder {
+        private FlinkContainersSettings flinkContainersSettings =
+                FlinkContainersSettings.defaultConfig();
+        private TestcontainersSettings testcontainersSettings =
+                TestcontainersSettings.defaultSettings();
+
+        private Builder() {}
+
+        public Builder withFlinkContainerSettings(FlinkContainersSettings flinkContainerSettings) {
```

Comment:
It looks like the naming of methods and classes are not aligned (withFlink**Container**Settings vs. Flink**Containers**Settings / with**Testcontainer**Settings vs. **Testcontainers**Settings)